templates/next/pages/foos/[id]/edit.tsx

```diff
@@ -1,30 +1,56 @@
-import { NextComponentType, NextPageContext } from 'next';
-import { Form } from '../../../components/{{{lc}}}/Form';
-import { {{{ucf}}} } from '../../../types/{{{ucf}}}';
-import { fetch } from '../../../utils/dataAccess';
+import { GetStaticPaths, GetStaticProps, NextComponentType, NextPageContext } from "next";
+import { Form } from "../../../components/{{{lc}}}/Form";
+import { {{{ucf}}} } from "../../../types/{{{ucf}}}";
+import { fetch } from "../../../utils/dataAccess";
 import Head from "next/head";
+import DefaultErrorPage from "next/error";
 
 interface Props {
   {{{lc}}}: {{{ucf}}};
 };
 
 const Page: NextComponentType<NextPageContext, Props, Props> = ({ {{{lc}}} }) => {
+  if (!{{{lc}}}) {
+    return <DefaultErrorPage statusCode={404} />;
+  }
+
   return (
     <div>
       <div>
         <Head>
-          <title>{ {{{lc}}} && `Edit {{{ucf}}} ${ {{~lc}}['@id']}`}</title>
+          <title>{ {{{lc}}} && `Edit {{{ucf}}} ${ {{~lc}}['@id'] }` }</title>
         </Head>
       </div>
-      <Form {{{lc}}}={ {{{lc}}} }/>
+      <Form {{{lc}}}={ {{{lc}}} } />
     </div>
   );
 };
 
-Page.getInitialProps = async ({ asPath }: NextPageContext) => {
-  const {{{lc}}} = await fetch(asPath.replace( '/edit', ''));
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  return {
+    props: {
+      {{{lc}}}: await fetch(`/{{{name}}}/${params.id}`),
+    },
+    revalidate: 1,
+  }
+}
 
-  return { {{{lc}}} };
-};
+export const getStaticPaths: GetStaticPaths = async () => {
+  try {
+    const response = await fetch("/{{{name}}}");
+
+    return {
+      paths: response.data["hydra:member"].map(({{{lc}}}) => `${ {{~lc}}['@id'] }/edit`),
```

Comment:
What about the second page etc? Shouldn't we not pre-generate this?

Reply:
Fixed in the last commit, but not really sure about what I've done :laughing: